File: JavaScript/Searches/BinarySearch.js
const prompt = require('prompt-sync')({sigint:true}); // for input
const n = prompt("Enter no of elements: ")
let arr = [];
console.log("Enter elements in accending order : --| ");
for(i=0;i<n;i++){
    const a = prompt();
    arr.push(parseInt(a));
}// parseInt() => type castes string to int; 
const x = prompt("Enter the element to be found: ");
let found = false;
let start=0;
let mid;
let end = n;
while(true){
    mid = Math.floor((start+end)/2);
    if(parseInt(x) === arr[mid]){
        console.log("Found at index "+mid+" !!");
        found = true;
        break;
    }
    if(start = mid){
        break;
    }
    else if( parseInt(x) > arr[mid]){
        start = mid;
    }
    else{
        end = mid;
    }
}
if(!found){
    console.log("Element not found !!");
}

File: Java/Searches/BinarySearch.java
import java.util.Scanner;
public class BinarySearch {
    public static void main(String[] args) {
        int n;
        int x;
        int i;
        int start , mid ,  end;
        boolean found = false;
        Scanner Sc = new Scanner(System.in);
        System.out.print("Enter no of elements: ");
        n = Sc.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter elements in accending order --| ");
        for(i=0;i<n;i++){
            arr[i] = Sc.nextInt();
        }
        System.out.print("Enter number to be found: ");
        x = Sc.nextInt();
        start = 0;
        end = n;
        while(true){
            mid = (start+end)/2;
            if(x == arr[mid]){
                    System.out.println("Found at index "+mid+" !!");
                    found = true;
                    break;
            } 
            if(start == mid){
                break;
            }
            else if( x > arr[mid]){
                    start = mid;
            }
            else{
                    end = mid;
           }
        }
        if(!found){
            System.out.println("Not found !!");
        }
    }
}